### ISSUE TITLE
Solution Upgrade failure

### ISSUE THREAD

**Issue by don-matese** [WaelHamze/dyn365-ce-vsts-tasks#224]
Since the weekend applying the solution upgrade has failed - this has been working fine for weeks and no changes have been made to the deployment process.  

After 36 seconds (running async) the error received is 

`Solution upgrade Failed. Error: Solution still exists after upgrade: Solution_Upgrade`

Looking in the solution history screen I can see references to 80040216 - Unexpected error (80040216) when importing solution in Dynamics 365.  Trying to apply the upgrade manually results in the same error and the upgrade remains unapplied.

https://support.microsoft.com/en-us/help/4457962/unexpected-error-80040216-when-importing-solution-in-dynamics-365

**Comment by AlexSpodkiewicz** [comment]
Yep I am getting a similar issue in the DeleteandPromote System job getting this error:

Exception retrieving custom activity info using SandboxPlugin - Unhandled exception: 
Exception type: Microsoft.Crm.CrmException
Message: Unable to retrieve customActivityInfo using RetrieveCustomActivityInfoWithSandboxPlugin.
   at Microsoft.Crm.Workflow.ObjectModel.CustomActivityInfoMetadataBase.GetCustomActivityInfo(ActivityInfo activityInfo, Int32 isolationMode, Guid pluginAssemblyId)
-- End stack trace --


Also in Solution History where you can see its failed the Error Code is showing as 8004505A

**Comment by dartamon** [comment]
Same upgrade error, when redeploying it goes away. No particular reason.

**Comment by WaelHamze** [comment]
@don-matese looks like this is due to something in the backend. I don't think there is an issue with the task. Did you try to raise a case with MS?

**Comment by don-matese** [comment]
Yes. There was an issue with environment variables following one of the
weekly updates to the platform. MS had to run an update script to
overcome the problem.

**Comment by Rhirhi229** [comment]
> @don-matese looks like this is due to something in the backend. I don't think there is an issue with the task. Did you try to raise a case with MS?

Hiya, I know this is an old thread but we are having the same issue. Is there an easy way to check any missing references / dependencies or is this issue occuring always linked to something MS needs to fix?
